File: ClientApp/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
    gender: string;
    minAge = 10;
    maxAge = 99;
    pageNumber = 1;
    pageSize = 5;
    orderBy = 'lastActive';
    knownAs = 'constestant';

    constructor(user: User) {
        this.gender = user.gender === 'female' ? 'male' : 'female';
        this.knownAs =user.knownAs === 'contestant' ? 'member' :'contestant';
    }

}

File: API/Helpers/UserParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Helpers
{
    public class UserParams : PaginationsParams
    {
        public string? CurrentUsername { get; set; }
        public string? Gender { get; set; }
        public int MinAge { get; set; } = 8;
        public int MaxAge { get; set; } = 150;
        public int RegistrationCount { get; set; } = 50;
        //srting 
        public string OrderBy { get; set; } = "lastActive";

        public string? KnownAs { get; set; } = "contestant";

        public bool? IsActive { get; set; } = true;
    }
}